app/alerts.py

```diff
@@ -201,55 +194,54 @@ def build_vision_polygon(site_lat, site_lon, yaw, opening_angle, dist_km):
+    All these inputs are instantiated in the main.py file via a POST from the API.
 
     In the base case, the function returns:
 
-    - the URL address of the image to be displayed on the left of the map;
-    - the alert button (banner above the map);
-    - the alert markers displayed on the map.
+    - the urls addresses of the image to be displayed on the left of the map
+    - the new_alert_button
+    - the alert markers displayed on the map
+    - new navbar_color as in alert mode
+    - new navbar title
 
     But if the style of map in place is 'risks', we don't want to display neither the alert markers.
     So in this case, the third output of the function is a void string.
     """
 
-    # Fetching alert status and reusable metadata
-    alert_lat = alert_metadata["lat"]
-    alert_lon = alert_metadata["lon"]
-    alert_id = str(alert_metadata["id"])
-
-    if alert_status == 1:
-        # Building the button that allows users to zoom towards the alert marker
-        alert_button = dbc.Button(
-            children="Départ de feu, cliquez-ici !",
-            color="danger",
-            block=True,
-            id=f'alert_button_{map_style}'
-        )
-
-        # Format of the alert marker icon
-        icon = {
-            "iconUrl": '../assets/pyro_alert_icon.png',
-            "iconSize": [50, 50],        # Size of the icon
-            "iconAnchor": [25, 45],      # Point of the icon which will correspond to marker's and popup's location
-            "popupAnchor": [0, -20]      # Point from which the popup should open relative to the iconAnchor
-        }
-
-        # Building the list of alert markers to be displayed
-        alerts_markers = [dl.Marker(
-            id="alert_marker_{}".format(alert_id),   # Setting a unique id for each alert marker
+    # Changing the navabar color and title
+    navbar_color = "#f34848"
+    navbar_title = "Départs de feux détectés"
+
+    # Format of the alert marker icon
+    icon = {
+        "iconUrl": '../assets/pyro_alert_icon.png',
+        "iconSize": [50, 50],        # Size of the icon
+        "iconAnchor": [25, 45],      # Point of the icon which will correspond to marker's and popup's location
+        "popupAnchor": [0, -20]      # Point from which the popup should open relative to the iconAnchor
+    }
+
+    # Building the list of alert markers to be displayed
+    alerts_markers = []
+    all_alerts = pd.read_json(live_alerts)
+    all_events = all_alerts.drop_duplicates(['id', 'event_id']).groupby('event_id').head(1)  # Get unique events
+    for _, row in all_events.iterrows():
+        alert_id = str(row['event_id'])
+        alert_lat = row['lat']
+        alert_lon = row['lon']
+
+        alerts_markers.append(dl.Marker(
+            id={'type': 'alert_marker', 'index': alert_id},   # Setting a unique id for each alert marker
             position=(alert_lat, alert_lon),
             icon=icon,
             children=[dl.Popup(
```

Comment:
Thanks a lot for updating the IDs of the components of the marker popup! Although we probably kill the popup within the new alert workflow, right?

Reply:
Indeed, did that to handle and test multiple alerts markers